examples/CampaignManagement/AddCompleteCampaignsUsingMutateJob.php

```diff
@@ -0,0 +1,486 @@
+ */
+
+namespace Google\Ads\GoogleAds\Examples\CampaignManagement;
+
+require __DIR__ . '/../../vendor/autoload.php';
+
+use GetOpt\GetOpt;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentNames;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentParser;
+use Google\Ads\GoogleAds\Lib\OAuth2TokenBuilder;
+use Google\Ads\GoogleAds\Lib\V1\GoogleAdsClient;
+use Google\Ads\GoogleAds\Lib\V1\GoogleAdsClientBuilder;
+use Google\Ads\GoogleAds\Lib\V1\GoogleAdsException;
+use Google\Ads\GoogleAds\Util\V1\ResourceNames;
+use Google\Ads\GoogleAds\V1\Services\AdGroupServiceClient;
+use Google\Ads\GoogleAds\V1\Common\ExpandedTextAdInfo;
+use Google\Ads\GoogleAds\V1\Common\KeywordInfo;
+use Google\Ads\GoogleAds\V1\Common\ManualCpc;
+use Google\Ads\GoogleAds\V1\Enums\AdGroupAdStatusEnum\AdGroupAdStatus;
+use Google\Ads\GoogleAds\V1\Enums\AdGroupCriterionStatusEnum\AdGroupCriterionStatus;
+use Google\Ads\GoogleAds\V1\Enums\AdGroupTypeEnum\AdGroupType;
+use Google\Ads\GoogleAds\V1\Enums\AdvertisingChannelTypeEnum\AdvertisingChannelType;
+use Google\Ads\GoogleAds\V1\Enums\BudgetDeliveryMethodEnum\BudgetDeliveryMethod;
+use Google\Ads\GoogleAds\V1\Enums\CampaignStatusEnum\CampaignStatus;
+use Google\Ads\GoogleAds\V1\Enums\KeywordMatchTypeEnum\KeywordMatchType;
+use Google\Ads\GoogleAds\V1\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V1\Errors\GoogleAdsFailure;
+use Google\Ads\GoogleAds\V1\Resources\Ad;
+use Google\Ads\GoogleAds\V1\Resources\AdGroup;
+use Google\Ads\GoogleAds\V1\Resources\AdGroupAd;
+use Google\Ads\GoogleAds\V1\Resources\AdGroupCriterion;
+use Google\Ads\GoogleAds\V1\Resources\Campaign;
+use Google\Ads\GoogleAds\V1\Resources\CampaignBudget;
+use Google\Ads\GoogleAds\V1\Resources\CampaignCriterion;
+use Google\Ads\GoogleAds\V1\Services\AdGroupAdOperation;
+use Google\Ads\GoogleAds\V1\Services\AdGroupCriterionOperation;
+use Google\Ads\GoogleAds\V1\Services\AdGroupOperation;
+use Google\Ads\GoogleAds\V1\Services\CampaignBudgetOperation;
+use Google\Ads\GoogleAds\V1\Services\CampaignCriterionOperation;
+use Google\Ads\GoogleAds\V1\Services\CampaignOperation;
+use Google\Ads\GoogleAds\V1\Services\CampaignServiceClient;
+use Google\Ads\GoogleAds\V1\Services\MutateJobResult;
+use Google\Ads\GoogleAds\V1\Services\MutateOperation;
+use Google\ApiCore\ApiException;
+use Google\Protobuf\BoolValue;
+use Google\Protobuf\Int64Value;
+use Google\Protobuf\StringValue;
+
+/**
+ * This example adds complete campaigns including campaign budgets, campaigns, ad groups and
+ * keywords using MutateJobService.
+ */
+class AddCompleteCampaignsUsingMutateJob
+{
+    const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+
+    const NUMBER_OF_CAMPAIGNS_TO_ADD = 2;
+    const NUMBER_OF_ADGROUPS_TO_ADD = 2;
+    const NUMBER_OF_KEYWORDS_TO_ADD = 5;
+    const POLL_FREQUENCY_SECONDS = 1;
+    const MAX_TOTAL_POLL_INTERVAL_SECONDS = 60;
+
+    const PAGE_SIZE = 1000;
+
+    /** @var int the negative temporary ID used in mutate job operations. */
+    private static $temporaryId = 0;
```

Comment:
It's a bit unclear here IMHO: if it's negative, why are you initializing it to 0?

Reply:
I changed to start from -1 instead.